pkg/cloud/externalconn/impl_registry.go

```diff
@@ -160,4 +248,14 @@ type TestingKnobs struct {
 // ModuleTestingKnobs is part of the base.ModuleTestingKnobs interface.
 func (t *TestingKnobs) ModuleTestingKnobs() {}
 
+// GetSkipCheckingExternalStorageConnection implements cloud.ExternalConnTestingKnobs.
+func (t *TestingKnobs) GetSkipCheckingExternalStorageConnection() func() bool {
```

Comment:
personally I think this is a little less readable than just accessing the TestingKnob bools directly. I might be missing something but I'd prefer if our if conditions looked like:

`if knobs.SkipCheckingExternalStorageConnection != nil && knobs.SkipCheckingExternalConnection()`

instead of

`if knobs.GetSkipCheckingExternalStorageConnection() != nil && knobs.GetSkipCheckingExternalStorageConnection()()`